src/engine/Service.cpp

```diff
@@ -107,6 +102,28 @@ ProtoResult Service::computeResult([[maybe_unused]] bool requestLaziness) {
         parsedServiceClause_.graphPatternAsString_.substr(openBracketPos + 1);
   }
 
+  try {
+    return computeResultImpl(requestLaziness);
+  } catch (const ad_utility::CancellationException&) {
+    throw;
+  } catch (const ad_utility::detail::AllocationExceedsLimitException&) {
+    throw;
```

Comment:
I think it would be easy 
(as a very last step to this PR)
to also test those two cases
(I think it should be possible to make the mocking getRequestImplFunction in the test throw these exceptions).
Then we have perfect test coverage.